presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -165,7 +165,8 @@ public Principal getUserPrincipal()
                         return;
                     }
                 }
-                catch (GSSException e) {
+                catch (Exception e) {
+                    LOG.debug(e, "Authentication failed for token %s", parts[1]);
```

Comment:
I don't think it's logged anywhere, and the exception will finally be caught by the web server and thrown back to the client as '500 internal error' with the original stack trace(at least that's what I observed from the test). so I agree we can rewrite the propagated exception to include the token.